src/mgmt/tests/srv_drpc_tests.c

```diff
@@ -925,26 +939,202 @@ test_drpc_pool_set_prop_success(void **state)
+	assert_int_equal(get_prop_resp->properties[0]->number, exp_prop_nr);
+	assert_int_equal(get_prop_resp->properties[0]->numval, exp_prop_val);
+
+	mgmt__pool_get_prop_resp__free_unpacked(get_prop_resp, NULL);
+}
+
+static void
+expect_drpc_pool_get_prop_str_success(Drpc__Response *resp,
+				      uint32_t exp_prop_nr,
+				      char *exp_prop_val)
+{
+	Mgmt__PoolGetPropResp *get_prop_resp = NULL;
+
+	assert_int_equal(resp->status, DRPC__STATUS__SUCCESS);
+	assert_non_null(resp->body.data);
+
+	get_prop_resp = mgmt__pool_get_prop_resp__unpack(NULL,
+			resp->body.len, resp->body.data);
+	assert_non_null(get_prop_resp);
+	assert_int_equal(get_prop_resp->n_properties, 1);
+	assert_int_equal(get_prop_resp->properties[0]->number, exp_prop_nr);
+	assert_string_equal(get_prop_resp->properties[0]->strval, exp_prop_val);
+
+	mgmt__pool_get_prop_resp__free_unpacked(get_prop_resp, NULL);
+}
+
+static void
+test_drpc_pool_get_prop_bad_uuid(void **state)
+{
+	Drpc__Call		call = DRPC__CALL__INIT;
+	Drpc__Response		resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolGetPropReq	req = MGMT__POOL_GET_PROP_REQ__INIT;
+
+	req.uuid = "wow this won't work";
+	req.properties = alloc_prop_msg_list(1);
+	req.n_properties = 1;
+	req.properties[0]->number = 1; /* doesn't matter */
+	setup_pool_get_prop_drpc_call(&call, &req);
+
+	ds_mgmt_drpc_pool_get_prop(&call, &resp);
+
+	expect_drpc_pool_get_prop_resp_with_error(&resp, -DER_INVAL);
+
+	free_prop_msg_list(req.properties, req.n_properties);
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+}
+
+static void
+test_drpc_pool_get_prop_num_success(void **state)
+{
+	Drpc__Call		call = DRPC__CALL__INIT;
+	Drpc__Response		resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolGetPropReq	req = MGMT__POOL_GET_PROP_REQ__INIT;
+	int			prop_number = DAOS_PROP_PO_SPACE_RB;
+	uint64_t		prop_val = 42;
+
+	ds_mgmt_pool_get_prop_out = daos_prop_alloc(1);
+	assert_non_null(ds_mgmt_pool_get_prop_out);
+	ds_mgmt_pool_get_prop_out->dpp_entries[0].dpe_type = prop_number;
+	ds_mgmt_pool_get_prop_out->dpp_entries[0].dpe_val = prop_val;
+
+	req.uuid = TEST_UUID;
+	req.properties = alloc_prop_msg_list(1);
+	req.n_properties = 1;
+	req.properties[0]->number = prop_number;
+	setup_pool_get_prop_drpc_call(&call, &req);
+
+	ds_mgmt_drpc_pool_get_prop(&call, &resp);
+
+	expect_drpc_pool_get_prop_num_success(&resp, prop_number, prop_val);
+
+	free_prop_msg_list(req.properties, req.n_properties);
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+}
+
+static void
+test_drpc_pool_get_prop_str_success(void **state)
+{
+	Drpc__Call		call = DRPC__CALL__INIT;
+	Drpc__Response		resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolGetPropReq	req = MGMT__POOL_GET_PROP_REQ__INIT;
+	int			prop_number = DAOS_PROP_PO_LABEL;
+	char			*prop_val = "foo";
+
+	ds_mgmt_pool_get_prop_out = daos_prop_alloc(1);
+	assert_non_null(ds_mgmt_pool_get_prop_out);
+	ds_mgmt_pool_get_prop_out->dpp_entries[0].dpe_type = prop_number;
+	D_STRNDUP(ds_mgmt_pool_get_prop_out->dpp_entries[0].dpe_str,
+		  prop_val, 4);
```

Comment:
@mjmac sorry, i change the value above to 4 to fix the build issue on fedora, but it sounds like it broke the unit test.